Add option to suppress warning when opening big files

DerivativeOfLog7 opened the pull request:
Add an option to suppress the >=2GB file warning. Won't show on a x86 build. This is mostly useful when leaving npp searching inside files unattended as the warning could pop up and halt all progress until acted upon
![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/100133857/dfbbe4b3-4827-4e42-adcc-2c33deb08b6f)

(the original PR had 4 👍 reacts)

alankilborn commented:
Just a thought: Maybe instead of on the `MISC.` subpage, this could go on the `Performance` subpage.  It isn't an exact fit there, but if I were looking for it, my first thought might be that subpage...

DerivativeOfLog7 commented:
> Just a thought: Maybe instead of on the `MISC.` subpage, this could go on the `Performance` subpage. It isn't an exact fit there, but if I were looking for it, my first thought might be that subpage...

Made it on a new branch: https://github.com/DerivativeOfLog7/notepad-plus-plus/tree/alt_placement

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/100133857/e36d72a1-2bde-4a48-bd23-e8a719f8b28a)

Positioning can of course be adjusted

donho commented on PowerEditor/src/ScintillaComponent/Buffer.cpp:
```diff
@@ -1483,21 +1483,23 @@ bool FileManager::loadFileData(Document doc, int64_t fileSize, const TCHAR * fil
 		else // x64
 		{
+			NppGUI& nppGui = NppParameters::getInstance().getNppGUI();
+			if (!nppGui._suppress2GBWarning) {
```
Please respect coding convention in Notepad++ :
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#general

donho commented on PowerEditor/src/WinControls/Preference/preference.rc:
```diff
@@ -23,18 +23,18 @@
 #endif
 
-IDD_PREFERENCE_BOX DIALOGEX 0, 0, 580, 235
+IDD_PREFERENCE_BOX DIALOGEX 0, 0, 580, 250
```
RESTORE ALL THE VALUE TO 235.

DerivativeOfLog7 replied:
Already done on the fork

donho commented:
@alankilborn 
> Just a thought: Maybe instead of on the MISC. subpage, this could go on the Performance subpage. 

For one of rare times, I agree with you totally! :)

This option, if it's not hidden, it should be in Performance subpage.

@DerivativeOfLog7 
And it should be like:

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/90293/a8c1c075-cf22-409e-a829-8b2ea9caa678)

donho reviewed (changes requested):
1. Please consider suggestions below.
2. Please consider https://github.com/notepad-plus-plus/notepad-plus-plus/pull/14053#issuecomment-1694495210

alankilborn commented:
>For one of rare times, I agree with you totally! :)

Well...really...why can't we make this happen more often??  :-)

DerivativeOfLog7 commented:
Moved the checkbox and fixed the code style

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/100133857/46517122-9e64-4701-b777-ced5a1608eda)

DerivativeOfLog7 commented:
I assume other variables added for the setting should be rearranged to be near the performance settings instead of the misc?

DerivativeOfLog7 commented:
Would it be appropriate to add a note stating that the setting affects search in files too?

alankilborn commented:
> Would it be appropriate to add a note stating that the setting affects search in files too?

A note to that effect can be added to the user manual description for the checkbox.  BTW, my personal expectation (which is meaningless) is that developers that get changes to the code accepted also write the user manual text.  (When I do a code change, I also provide the user manual change)

donho commented:
![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/90293/bc959587-a6d8-4335-945a-5ae00409b8cd)

donho commented:
> 11. Finally, please test your pull requests, at least once.

Ref: https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#guidelines-for-pull-requests

DerivativeOfLog7 commented:
> > 11. Finally, please test your pull requests, at least once.
> 
> Ref: https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#guidelines-for-pull-requests

Sorry... it was working when I first added it; I just forgot to reword the settings cpp after moving it to the performance tab
